src/attacks.rs

```diff
@@ -14,15 +22,214 @@ pub enum DepthReduceSet {
     /// Parameter  is size of the resulting G-S graph desired
     ValiantAB16(usize),
     /// depth of the resulting G-S graph and some specific parameters
-    Greedy(usize, GreedyParams),
+    GreedyDepth(usize, GreedyParams),
+    /// Variation of Greedy attack that has as `target` the resulting size of set S.
+    GreedySize(usize, GreedyParams),
 }
 
 pub fn depth_reduce(g: &mut Graph, drs: DepthReduceSet) -> HashSet<usize> {
     match drs {
         DepthReduceSet::ValiantDepth(_) => valiant_reduce(g, drs),
         DepthReduceSet::ValiantSize(_) => valiant_reduce(g, drs),
         DepthReduceSet::ValiantAB16(_) => valiant_reduce(g, drs),
-        DepthReduceSet::Greedy(target, p) => greedy_reduce(g, target, p),
+        DepthReduceSet::GreedyDepth(_, _) => greedy_reduce(g, drs),
+        DepthReduceSet::GreedySize(_, _) => greedy_reduce(g, drs),
+    }
+}
+
+/// Target of an attack, either the depth should be smaller than `Depth` or
+/// the exclusion set `S` should have a size bigger than `Size` (actually
+/// we want to hit a target as close as possible as those thresholds). All
+/// targets are specified in relation to the size of the graph `G` (not to
+/// be confused with the target size of set `S`).
+// FIXME: Overlapping with `DepthReduceSet` internal values, this should
+//  replace that.
+#[derive(Debug)]
+pub enum AttackTarget {
+    Depth(f64),
+    Size(f64),
+}
+
+/// Range of targets to try (to find the optimum value) from `start`, increasing
+/// by `interval` until `end` is reached or surpassed.
+// FIXME: Using this instead of `std::ops::Range<f64>` because Rust correctly
+//  doesn't allow iterating over floating point values but there is probably
+//  an easier way than coding this from scratch.
+// FIXME: Assert range validity in the struct itself instead of on the caller
+//  (`attack`).
+#[derive(Debug)]
+pub struct TargetRange {
+    pub start: f64,
+    pub interval: f64,
+    pub end: f64,
+}
+
+#[derive(Debug)]
+pub struct AttackProfile {
+    pub runs: usize,
+    pub target: AttackTarget,
+    pub range: TargetRange,
+    pub attack: DepthReduceSet,
+}
+
+impl AttackProfile {
+    /// Build a default profile from an attack type that has only one run
+    /// in a range of a single value (to make it compatible with previous
+    /// uses of `attack`).
+    // FIXME: We shouldn't need the `graph_size` (or the graph for that
+    // matter), but this is accommodating previous uses of `attack` (which
+    // should be refactored entirely and this method removed or reworked).
+    pub fn from_attack(attack: DepthReduceSet, graph_size: usize) -> Self {
+        let graph_size = graph_size as f64;
+        let target = match attack {
+            DepthReduceSet::ValiantDepth(depth) => AttackTarget::Depth(depth as f64 / graph_size),
```

Comment:
Maybe this is related to the FIXME just above but not sure I understand why do you give the fraction rather than the absolute number here ?